standards/code-style.adoc

```diff
@@ -338,14 +342,59 @@ if (condition) {
 }
 ----
 
-This is __bad__:
-
 [source, kotlin]
 ----
 if (condition)
     body()  // bad!
 ----
 
+Braces are not required for `when` branches and `if` statement bodies if they meet ALL
+the following conditions:
+
+. there are no `else if`/`else` branches
+. the conditional and the branch contain at most a single predicate
+. it fits on a single line
```

Comment:
Do we also want to restrict this to only conditionals that return from a method? For example, do we want to discourage or encourage the following?
`if (input.isEmpty) value = "Placeholder"`

Reply:
I think we definitely want to discourage that code (in favor of an expression), but I don't think we have to explicitly say that you have to return.  Something like this would be valid:
````kotlin
if (input.isEmpty) showPlaceholder()
// continue...
````